server, ui: remove txn_fingerprint_id in stmts overview query, stop aggregating on UI

xinhaoz opened the pull request:
See individual commits.

Fixes: #99708
Fixes: #99390

https://www.loom.com/share/d1a3a45f3c08425daab3a1fb9af01e37

xinhaoz committed:
ui: stop aggregating fingerprints response on UI

In the combined stats EP, we group entries on both fingerprint_id
and app_name for the overview page queries. The client has been
grouping only on fingerprint_id for both overview pages. With the
introduction of the limit param, this mismatch in aggregation criteria
can lead to fewer than the requested rows returned from the server.

This commit ensures that the UI does no additional aggregation with
the payload from the server. In the next commit, we will provide the
app name of the selected row when routing to statement fingerprint
details from the overview page, so that we only req stats for the
app of the selected row.

As a part of this change, we create a common combiner body for the stmts
selector in both db-console and cluster-ui to reduce duplicated code.
Relevant aggregation code on the client has also been removed.

Part of: #99708
Fixes: #99390

Release note (ui change): By default, we now show the `Application Name`
column in the fingerprints overview pages. Stmts and txns fingerprints
will be displayed per application on the overview pages rather than
grouped into a single fingerprint id.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/99719)
<!-- Reviewable:end -->

xinhaoz committed:
server: stop grouping stmts by transaction_fingerprint_id

Previously, the query for the stmt fingerprints overview page
grouped stmts on transaction_fingerprint_id. We should remove
this grouping criteria since we do not care about the txn at
this page level, and including it means we can't return as many
unique stmt fingerprint ids on the page. To view txn specific
stats / stmts for a txn, users can use the txn fingerprints page.
We add the distinct txn_fingerprint_ids for the grouped stmts as
a new field for testing purposes. Note that the
`transaction_fingerprint_id` in the key_data field for stmts returned
by the CombinedStatements api will now be blank.

New proto fields:
- txn_fingerprint_ids (for StatementsResponse_CollectedStatementStatistics)

Part of: #99708

Release note: None

xinhaoz committed:
ui: use app name of selected row when fetching fingerprint details

Previously, the fingerprint details pages would show the details
aggregated by the selected app names (from the filter, or aggregated
across all apps if none were selected). Since we no longer perform
any aggregation in the overview pages we now always display each
fingerprint attached to a single app for each row. When going to
the details page via clicking a row from the overview page, we
should fetch details only for the selected app.

Note that the txn fingerprint details page simply looks for a
corresponding entry from the overview response, thus we change
that search criteria to incude the app name of the selected row.

The app name has also been included in  the link to stmt details
from insights details.

Release note (ui change): When going from the fingerprint overview
pages or the insight details pages to the fingerprint details
page, the details page will fetch data for the stmt with the
provided application name. For overview pages, this is the app
name of the selected row. For insight details, it is the app of
the execution that generated the insight.

Part of: #99708

j82w commented:
*[`pkg/server/combined_statement_stats.go` line 374 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRhJb7Q2qUAORHSa6r4:-NRhJb7Q2qUAORHSa6r5:b-3lppjx) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/server/combined_statement_stats.go#L374)):*
> ```Go
>     transaction_fingerprint_id,
>     app_name,
>     metadata
> ```

Shouldn't metadata still be in the group by? It has the db, query, implicit, etc... which I think we still want to be unique. 
Example json:
{"db": "", "distsql": false, "failed": false, "fullScan": false, "implicitTxn": false, "query": "SELECT username FROM system.users WHERE username = $1", "querySummary": "SELECT username FROM system.users", "stmtTyp": "TypeDML", "vec": true}


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> Reviewed 1 of 26 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w)

___
*[`pkg/server/combined_statement_stats.go` line 374 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRhJb7Q2qUAORHSa6r4:-NRhTB417TnUUat5gMjB:b-pbvczt) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/server/combined_statement_stats.go#L374)):*
<details><summary><i>Previously, j82w (Jake) wrote…</i></summary><blockquote>

Shouldn't metadata still be in the group by? It has the db, query, implicit, etc... which I think we still want to be unique. 
Example json:
{"db": "", "distsql": false, "failed": false, "fullScan": false, "implicitTxn": false, "query": "SELECT username FROM system.users WHERE username = $1", "querySummary": "SELECT username FROM system.users", "stmtTyp": "TypeDML", "vec": true}
</blockquote></details>

we don't need it to be unique actually, some values will always be the same (db, query, etc), because is what the fingerprint is based on, so if can pick max min, it won't matter, but the values for vec, distsql, fullscan, etc can very, so there is actually a function that combines metadata that can be used. If you now do a proper combine you might also fix a bug where we're combining fingerprints with and without full scan, but on the overview we show just one of them


<!-- Sent from Reviewable.io -->

xinhaoz commented:
*[`pkg/server/combined_statement_stats.go` line 374 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRhJb7Q2qUAORHSa6r4:-NRiLNNJ-PR152xuGlVn:bbnvrxo) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/server/combined_statement_stats.go#L374)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

we don't need it to be unique actually, some values will always be the same (db, query, etc), because is what the fingerprint is based on, so if can pick max min, it won't matter, but the values for vec, distsql, fullscan, etc can very, so there is actually a function that combines metadata that can be used. If you now do a proper combine you might also fix a bug where we're combining fingerprints with and without full scan, but on the overview we show just one of them
</blockquote></details>

Done. Do we need to do the same for transactions?


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w)

___
*[`pkg/server/combined_statement_stats.go` line 374 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRhJb7Q2qUAORHSa6r4:-NRk2ZDb2ezY2Iw3TiiF:b9pkknd) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/server/combined_statement_stats.go#L374)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Done. Do we need to do the same for transactions?
</blockquote></details>

they behaviour needs to be consistent on both pages


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
Looks like you still have some tests to fix

> Reviewed 3 of 26 files at r2, 26 of 26 files at r5, 14 of 14 files at r6, 12 of 12 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-:-NRk5H-S-tlskg8UoNuK:b-p1nhyk)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w and @xinhaoz)

___
*[`-- commits` line 58 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk7aS4-pEVN7B5lP4w):*
I'm not seeing any changes on the stmt details endpoint to filter out just by that app name? is that correct?
___
*[`pkg/server/combined_statement_stats.go` line 472 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk4NEoDG-F6ssYMlBl:-NRk4NEp4IoXT2uh31gv:batscc7) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ae73cbc10235d7434861111c1bdaa678f995b02/pkg/server/combined_statement_stats.go#L472)):*
> ```Go
>     max(aggregated_ts) as aggregated_ts,
>     fingerprint_id,
>     max(metadata),
> ```

here you can combine too
___
*[`pkg/server/combined_statement_stats.go` line 562 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk4U2f2lmAZnMmbrA9:-NRk4U2f2lmAZnMmbrAA:bmy6h4) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ae73cbc10235d7434861111c1bdaa678f995b02/pkg/server/combined_statement_stats.go#L562)):*
> ```Go
>     fingerprint_id,
>     transaction_fingerprint_id,
>     max(metadata),
> ```

you will need to combine here too
___
*[`pkg/server/serverpb/status.proto` line 1599 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk4qJm4NiY69NoBQKk:-NRk4qJm4NiY69NoBQKl:bh3fvp1) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ae73cbc10235d7434861111c1bdaa678f995b02/pkg/server/serverpb/status.proto#L1599)):*
> ```protobuf
>     cockroach.sql.StatementStatistics stats = 2 [(gogoproto.nullable) = false];
>     // In 23.1 we expect the response to only group on fingerprint_id in the overview page.
>     // We now return the aggregated list of unique app names, leaving the app field in the key empty.
> ```

is this comment what you meant here? is on top of txn fingerprint but talking about app names
___
*[`pkg/ui/workspaces/cluster-ui/src/selectors/common.ts` line 37 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk5R124uyljBz6O9Py:-NRk5R13Dt2IybzVTm22:b-eysv4g) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e572313026e4a88097f2571f717823d4fc3ee97/pkg/ui/workspaces/cluster-ui/src/selectors/common.ts#L37)):*
> ```XML
> 
> export const selectExecutionID = (
>   _state: unknown,
> ```

we don't know the type here?
___
*[`pkg/ui/workspaces/cluster-ui/src/statementsPage/statementsPageConnected.tsx` line 96 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk5unW-tZoIeBQJggA:-NRk5unW-tZoIeBQJggB:b-qq675r) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e572313026e4a88097f2571f717823d4fc3ee97/pkg/ui/workspaces/cluster-ui/src/statementsPage/statementsPageConnected.tsx#L96)):*
> ```XML
>       fingerprintsPageProps: {
>         ...props,
>         apps: selectStmtsAllApps(state.adminUI.statements?.data),
> ```

can you add `adminUI?.statements...` ? The first time a cluster is created is takes a little while for the adminUI to be created and the page might crash
___
*[`pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx` line 178 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6Ce1-G8BS_9eSL3u:-NRk6Ce1-G8BS_9eSL3v:b-62e5i8) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e572313026e4a88097f2571f717823d4fc3ee97/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx#L178)):*
> ```XML
>         stmt.applicationName?.length > 0 ? stmt.applicationName : unset,
>       sort: (stmt: AggregateStatistics) => stmt.applicationName,
>       showByDefault: true,
> ```

the `showByDefault` defualt value is already `true`, so you only add this line if the value is false. You can remove this.
___
*[`pkg/ui/workspaces/cluster-ui/src/transactionsPage/transactionsPage.tsx` line 440 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6NyF9NsfR2xF9Q-P:-NRk6NyF9NsfR2xF9Q-Q:b-73n4a4) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/cluster-ui/src/transactionsPage/transactionsPage.tsx#L440)):*
> ```XML
>     ).map(t => ({
>       stats_data: t.stats_data,
>       node_id: t.node_id,
> ```

why you removed the node_id?
___
*[`pkg/ui/workspaces/cluster-ui/src/transactionsTable/transactionsTable.tsx` line 178 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6W1640w8rwhbsxHA:-NRk6W1640w8rwhbsxHB:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e572313026e4a88097f2571f717823d4fc3ee97/pkg/ui/workspaces/cluster-ui/src/transactionsTable/transactionsTable.tsx#L178)):*
> ```XML
>       sort: (item: TransactionInfo) => item.stats_data?.app,
> 
>       showByDefault: true,
> ```

ditto
___
*[`pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.spec.ts` line 486 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6fj77jVoXyjaBIQh:-NRk6fj77jVoXyjaBIQi:b-8f5zth) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.spec.ts#L486)):*
> ```XML
>     });
>   });
> });
> ```

looks like a lot of tests were removed, so can you add new ones for the new changes that were made?
___
*[`pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6o9h8zd3z0tAaNTs:-NRk6o9h8zd3z0tAaNTt:b-w23s6c) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx#L290)):*
> ```XML
> 
>     expect(result).toBe(1);
>   });
> ```

same comments from the tests from this file. We need to make sure the new behaviour is working as expected.


<!-- Sent from Reviewable.io -->

xinhaoz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w and @maryliag)

___
*[`-- commits` line 58 at r7](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk7aS4-pEVN7B5lP4w):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

I'm not seeing any changes on the stmt details endpoint to filter out just by that app name? is that correct?
</blockquote></details>

Yup that's correct. We don't make any changes to that process since it takes the app name from the URL and uses that to fetch data. I thought that as is was pretty useful (we have the functionality of aggregating by  multiple app names still, but we choose to provide only one).
The change i made in the 3rd commit ensures that we provide the single app name in the URL to the details page when coming from overview page / insights.
___
*[`pkg/server/combined_statement_stats.go` line 374 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRhJb7Q2qUAORHSa6r4:-NRkRxt49kFU7v8eVnRR:brns3w4) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/server/combined_statement_stats.go#L374)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

they behaviour needs to be consistent on both pages
</blockquote></details>

Ah I meant do we need to also create and use a merge_metadata fn for the txns query. I see in the other comment the answer is yes,but perhaps that change can be left to another PR since the function also needs to be created.
___
*[`pkg/server/combined_statement_stats.go` line 562 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk4U2f2lmAZnMmbrA9:-NRkSiBA-wvAoVCuWBmy:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ae73cbc10235d7434861111c1bdaa678f995b02/pkg/server/combined_statement_stats.go#L562)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

you will need to combine here too
</blockquote></details>

Done.
___
*[`pkg/server/serverpb/status.proto` line 1599 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk4qJm4NiY69NoBQKk:-NRkSjcX0IeR4eq4385L:b-fozez1) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ae73cbc10235d7434861111c1bdaa678f995b02/pkg/server/serverpb/status.proto#L1599)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

is this comment what you meant here? is on top of txn fingerprint but talking about app names
</blockquote></details>

Oops, no it is not. Fixed to reference txn_fingerprint_ids.
___
*[`pkg/ui/workspaces/cluster-ui/src/selectors/common.ts` line 37 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk5R124uyljBz6O9Py:-NRkSm2x8P_1tnsmZK45:byygyj4) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e572313026e4a88097f2571f717823d4fc3ee97/pkg/ui/workspaces/cluster-ui/src/selectors/common.ts#L37)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

we don't know the type here?
</blockquote></details>

Yeah we don't know if it's CC or DB-Console's state since this is used by both.
___
*[`pkg/ui/workspaces/cluster-ui/src/statementsPage/statementsPageConnected.tsx` line 96 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk5unW-tZoIeBQJggA:-NRkT3GP3ytfofG5DgcF:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e572313026e4a88097f2571f717823d4fc3ee97/pkg/ui/workspaces/cluster-ui/src/statementsPage/statementsPageConnected.tsx#L96)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

can you add `adminUI?.statements...` ? The first time a cluster is created is takes a little while for the adminUI to be created and the page might crash
</blockquote></details>

Done.
___
*[`pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx` line 178 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6Ce1-G8BS_9eSL3u:-NRkTBsoAyT4SrhPDoa7:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e572313026e4a88097f2571f717823d4fc3ee97/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx#L178)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

the `showByDefault` defualt value is already `true`, so you only add this line if the value is false. You can remove this.
</blockquote></details>

Done.
___
*[`pkg/ui/workspaces/cluster-ui/src/transactionsPage/transactionsPage.tsx` line 440 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6NyF9NsfR2xF9Q-P:-NRkTKpL0v3Mc1zQDogQ:b-keinys) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/cluster-ui/src/transactionsPage/transactionsPage.tsx#L440)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

why you removed the node_id?
</blockquote></details>

It's not populated on the server so I don't think it was doing anything.
___
*[`pkg/ui/workspaces/cluster-ui/src/transactionsTable/transactionsTable.tsx` line 178 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6W1640w8rwhbsxHA:-NRkTaap8ma7JMhSgKoY:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5e572313026e4a88097f2571f717823d4fc3ee97/pkg/ui/workspaces/cluster-ui/src/transactionsTable/transactionsTable.tsx#L178)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.
___
*[`pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.spec.ts` line 486 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6fj77jVoXyjaBIQh:-NRkTdFE4AN0Q_bnX2l6:b8wd517) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.spec.ts#L486)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

looks like a lot of tests were removed, so can you add new ones for the new changes that were made?
</blockquote></details>

These removed tests are just for the aggregation step on the UI which was removed completely. We're still continuing to test the selectStmts and selectApps selectors which were both combined into one combiner fn each (to reuse across db-console and cluster-ui). The existing tests should verify that behavior is still working in terms of filtering etc. For the aggregation removals, do you think it would be useful to add some checks to confirm there is no aggregation being done anymore or some other test?
___
*[`pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6o9h8zd3z0tAaNTs:-NRkU5E58aYyKecueeB5:b-3nczvb) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx#L290)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

same comments from the tests from this file. We need to make sure the new behaviour is working as expected.
</blockquote></details>

The new behaviour is just returning exactly the field without any aggregation. Should I add some tests where we have the same fingerprint id just to make sure no aggregation is being done (as in the other comment).


<!-- Sent from Reviewable.io -->

xinhaoz commented:
*[`pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6o9h8zd3z0tAaNTs:-NRnETQbCozOnKvyaFtN:b-w86d2q) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx#L290)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

The new behaviour is just returning exactly the field without any aggregation. Should I add some tests where we have the same fingerprint id just to make sure no aggregation is being done (as in the other comment).
</blockquote></details>

Actually, scratch that for this comment what I said is incorrect after the updated changes. -- This needs to be reverted since we're still grouping on app_name in the server. I'll make those changes and re-introduce the testing here.


<!-- Sent from Reviewable.io -->

xinhaoz committed:
ui: remove `totalFingerprints` prop from statement page

This commit removes the unused prop `totalFingerprints` and its
relevant selectors.

Epic: none

Release note: None

xinhaoz commented:
I ended up removing the totalFingerprints logic since it wasn't actually being used. Also added some tests to ensure grouping on fingerprints isn't happening anymore via the selectStatements function.

maryliag reviewed (commented):
> Reviewed 7 of 22 files at r3, 16 of 16 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w and @xinhaoz)

___
*[`pkg/server/combined_statement_stats.go` line 374 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRhJb7Q2qUAORHSa6r4:-NS7uRkmAwnb23gloItr:b-ag69mg) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/server/combined_statement_stats.go#L374)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Ah I meant do we need to also create and use a merge_metadata fn for the txns query. I see in the other comment the answer is yes,but perhaps that change can be left to another PR since the function also needs to be created.
</blockquote></details>

You will need to rebase, there were big changes in this file
___
*[`pkg/server/combined_statement_stats.go` line 570 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS7uf496ntPh7UYtUn-:-NS7uf496ntPh7UYtUn0:bqgnbkj) ([raw file](https://github.com/cockroachdb/cockroach/blob/28f622e7582f49fde3bc2f7fa17175e9acb3961f/pkg/server/combined_statement_stats.go#L570)):*
> ```Go
>     transaction_fingerprint_id,
>     app_name
> `, whereClause)
> ```

Looks at the changes introduced on https://github.com/cockroachdb/cockroach/pull/100440, you will need to do similar to this select, where you use the combined view as a backup
___
*[`pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx` line 126 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS7wQjd1G2ifaTUi3Wh:-NS7wQjd1G2ifaTUi3Wi:by1qjf7) ([raw file](https://github.com/cockroachdb/cockroach/blob/65d9792252bef81310f9d5ed4d27e1fb93a32194/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx#L126)):*
> ```XML
>   const sampledExecStatsBarChartOptions = {
>     classes: defaultBarChartOptions.classes,
>     displayNoSamples: (d: ICollectedStatementStatistics) => {
> ```

when we have no samples (a lot of cases when trying to display execution stats), we want to show the no samples message.  Why this line was removed?
___
*[`pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTableContent.tsx` line 157 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS8-7Bn-peadz6uKT9u:-NS8-7Bn-peadz6uKT9v:b-vuqirc) ([raw file](https://github.com/cockroachdb/cockroach/blob/65d9792252bef81310f9d5ed4d27e1fb93a32194/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTableContent.tsx#L157)):*
> ```XML
>       );
>     },
>   nodeLink:
> ```

why is this removed?
___
*[`pkg/ui/workspaces/cluster-ui/src/transactionsTable/transactionsTable.tsx` line 105 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS7xS7qEITbmmUJXfir:-NS7xS7qEITbmmUJXfis:b3638np) ([raw file](https://github.com/cockroachdb/cockroach/blob/28f622e7582f49fde3bc2f7fa17175e9acb3961f/pkg/ui/workspaces/cluster-ui/src/transactionsTable/transactionsTable.tsx#L105)):*
> ```XML
>   const sampledExecStatsBarChartOptions = {
>     classes: defaultBarChartOptions.classes,
>     _: (d: TransactionInfo) => {
> ```

same questions is the other, is the value showing as 0 when there are no samples or the correct value of "no samples" ? 
___
*[`pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.ts` line 321 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS7yEpa-dTPlnhPPZSg:-NS7yEpb7bMvANc7eWzo:bhj8sgd) ([raw file](https://github.com/cockroachdb/cockroach/blob/28f622e7582f49fde3bc2f7fa17175e9acb3961f/pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.ts#L321)):*
> ```XML
>   stmt: ExecutionStatistics,
> ): string {
>   return statementKey(stmt) + stmt.txn_fingerprint_ids.toString();
> ```

add a `?` because older version won't have `txn_fingerprint_ids`
___
*[`pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6o9h8zd3z0tAaNTs:-NS7ygnzAkaiFxP5TP22:b-5ocvmn) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx#L290)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

Actually, scratch that for this comment what I said is incorrect after the updated changes. -- This needs to be reverted since we're still grouping on app_name in the server. I'll make those changes and re-introduce the testing here.
</blockquote></details>

Just making sure the tests are now correct


<!-- Sent from Reviewable.io -->

xinhaoz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w and @maryliag)

___
*[`pkg/server/combined_statement_stats.go` line 374 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRhJb7Q2qUAORHSa6r4:-NS8o61p-ZTqsPTx9wVv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/server/combined_statement_stats.go#L374)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

You will need to rebase, there were big changes in this file
</blockquote></details>

Done.
___
*[`pkg/server/combined_statement_stats.go` line 472 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk4NEoDG-F6ssYMlBl:-NS8oBWVE9x41tqTWN8i:b-tkppoi) ([raw file](https://github.com/cockroachdb/cockroach/blob/2ae73cbc10235d7434861111c1bdaa678f995b02/pkg/server/combined_statement_stats.go#L472)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

here you can combine too
</blockquote></details>

For txns it seems we can guarantee the metadata will be unique based on the txn_fingerprint_id. The only field added to it currently is stmt_fingerprint_ids.
___
*[`pkg/server/combined_statement_stats.go` line 570 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS7uf496ntPh7UYtUn-:-NS8o8Gm60_H47_Ez7Hg:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/28f622e7582f49fde3bc2f7fa17175e9acb3961f/pkg/server/combined_statement_stats.go#L570)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

Looks at the changes introduced on https://github.com/cockroachdb/cockroach/pull/100440, you will need to do similar to this select, where you use the combined view as a backup
</blockquote></details>

Done.
___
*[`pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx` line 126 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS7wQjd1G2ifaTUi3Wh:-NS8kvbN61rTwleCJaGF:bdhy10b) ([raw file](https://github.com/cockroachdb/cockroach/blob/65d9792252bef81310f9d5ed4d27e1fb93a32194/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTable.tsx#L126)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

when we have no samples (a lot of cases when trying to display execution stats), we want to show the no samples message.  Why this line was removed?
</blockquote></details>

For some reason my IDE was complaining that this field was unused. I put it back and added some typing to it to stop the lint warning.
___
*[`pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTableContent.tsx` line 157 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS8-7Bn-peadz6uKT9u:-NS8lkK107jb2913frCO:bf20gfm) ([raw file](https://github.com/cockroachdb/cockroach/blob/65d9792252bef81310f9d5ed4d27e1fb93a32194/pkg/ui/workspaces/cluster-ui/src/statementsTable/statementsTableContent.tsx#L157)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

why is this removed?
</blockquote></details>

This was not being used, so I removed it, I verified there's no use cases after the other ones were marked as unused incorrectly.
___
*[`pkg/ui/workspaces/cluster-ui/src/transactionsTable/transactionsTable.tsx` line 105 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS7xS7qEITbmmUJXfir:-NS8logw-lkUKGh8ypfW:bpwr5cy) ([raw file](https://github.com/cockroachdb/cockroach/blob/28f622e7582f49fde3bc2f7fa17175e9acb3961f/pkg/ui/workspaces/cluster-ui/src/transactionsTable/transactionsTable.tsx#L105)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

same questions is the other, is the value showing as 0 when there are no samples or the correct value of "no samples" ? 
</blockquote></details>

Same weird linting warning as above, but I added the typing and it seemed to fix the warning.
___
*[`pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.ts` line 321 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NS7yEpa-dTPlnhPPZSg:-NS8lt3t3WU8ch2eOeRS:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/28f622e7582f49fde3bc2f7fa17175e9acb3961f/pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.ts#L321)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

add a `?` because older version won't have `txn_fingerprint_ids`
</blockquote></details>

Done.
___
*[`pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx` line 290 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6o9h8zd3z0tAaNTs:-NS8m0Ec4yIn_crN55Qw:b-re1l0) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/db-console/src/views/statements/statements.spec.tsx#L290)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

Just making sure the tests are now correct
</blockquote></details>

Yup, removed the tests that no longer apply and added some to verify that aggregation is no longer occurring.


<!-- Sent from Reviewable.io -->

maryliag reviewed (commented):
> Reviewed 7 of 7 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w and @xinhaoz)

___
*[`pkg/server/combined_statement_stats.go` line 650 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NSfVbRR5Y6ryJkxihu8:-NSfVbRR5Y6ryJkxihu9:b-b4am9s) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0604fda71602c9490962da5e1b2b20ab9e4e14c/pkg/server/combined_statement_stats.go#L650)):*
> ```Go
> 	const expectedNumDatums = 5
> 
> 	query := fmt.Sprintf(queryFormat, "crdb_internal.statement_statistics_persisted", whereClause)
> ```

There are a few more changes being added [here](https://github.com/cockroachdb/cockroach/pull/100703/files#diff-d0be44319d1d1e310ff9b5b2680e7bfdf20f045d702802f659843e19dc3873f1R86), that is currently on bors queue, so once that gets is you need to add something similar to use 22_2 view
___
*[`pkg/server/combined_statement_stats.go` line 689 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NSfVmeb4Yhi3rzTsdHH:-NSfVmec3d1d17Uo5EL3:b-9fnv72) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0604fda71602c9490962da5e1b2b20ab9e4e14c/pkg/server/combined_statement_stats.go#L689)):*
> ```Go
> 		var row tree.Datums
> 		if row = it.Cur(); row == nil {
> 			return nil, errors.New("unexpected null row")
> ```

can you add a `... on collectStmtsForTxns` for the error messages here and the error below too? To make it easier to identify errors later on



_Code quote:_
```Go
collectStmtsForTxns
```
___
*[`pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.spec.ts` line 486 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6fj77jVoXyjaBIQh:-NSfWOiR-R2s5dl9_bLQ:bggkelt) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.spec.ts#L486)):*
<details><summary><i>Previously, xinhaoz (Xin Hao Zhang) wrote…</i></summary><blockquote>

These removed tests are just for the aggregation step on the UI which was removed completely. We're still continuing to test the selectStmts and selectApps selectors which were both combined into one combiner fn each (to reuse across db-console and cluster-ui). The existing tests should verify that behavior is still working in terms of filtering etc. For the aggregation removals, do you think it would be useful to add some checks to confirm there is no aggregation being done anymore or some other test?
</blockquote></details>

Just replying here that it does look like you added the tests to check the aggregation. It should be enough for these changes


<!-- Sent from Reviewable.io -->

xinhaoz reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @j82w and @maryliag)

___
*[`pkg/server/combined_statement_stats.go` line 650 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NSfVbRR5Y6ryJkxihu8:-NSf__oD-ts1M67WmlOA:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0604fda71602c9490962da5e1b2b20ab9e4e14c/pkg/server/combined_statement_stats.go#L650)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

There are a few more changes being added [here](https://github.com/cockroachdb/cockroach/pull/100703/files#diff-d0be44319d1d1e310ff9b5b2680e7bfdf20f045d702802f659843e19dc3873f1R86), that is currently on bors queue, so once that gets is you need to add something similar to use 22_2 view
</blockquote></details>

Done.
___
*[`pkg/server/combined_statement_stats.go` line 689 at r9](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NSfVmeb4Yhi3rzTsdHH:-NSf_bAe6t6800qglHqb:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/c0604fda71602c9490962da5e1b2b20ab9e4e14c/pkg/server/combined_statement_stats.go#L689)):*
<details><summary><i>Previously, maryliag (Marylia Gutierrez) wrote…</i></summary><blockquote>

can you add a `... on collectStmtsForTxns` for the error messages here and the error below too? To make it easier to identify errors later on
</blockquote></details>

Done.
___
*[`pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.spec.ts` line 486 at r6](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-NRk6fj77jVoXyjaBIQh:-NSf_e7_5eeHduesthbY:bgjudtt) ([raw file](https://github.com/cockroachdb/cockroach/blob/1e2ea17a3dbb4b4c8a2942105abc6712d90b477e/pkg/ui/workspaces/cluster-ui/src/util/appStats/appStats.spec.ts#L486)):*
> .
Awesome!


<!-- Sent from Reviewable.io -->

maryliag reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 1 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99719#-:-NSgK6-H54Cuj0J1YM6a:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @j82w)



<!-- Sent from Reviewable.io -->

xinhaoz commented:
TFTR!!!
bors r+
